.changeset/brown-bats-float.md

```diff
@@ -0,0 +1,10 @@
+---
+"@apollo/query-planner": patch
+"@apollo/composition": patch
+"@apollo/federation-internals": patch
+"@apollo/gateway": patch
+---
+
+Improves the performance of the code used to try to reuse query named fragments, thus improving query planning
```

Comment:
Add a note about the other issues being fixed?

Reply:
Good point. Re-worked the message to include them.